server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTreeBelowSG.java

```diff
@@ -177,14 +169,28 @@ public void init() throws IOException {
     File mtreeSnapshot = SystemFileFactory.INSTANCE.getFile(mtreeSnapshotPath);
     long time = System.currentTimeMillis();
     if (mtreeSnapshot.exists()) {
-      this.root = deserializeFrom(mtreeSnapshot).root;
-      logger.debug(
-          "spend {} ms to deserialize mtree from snapshot", System.currentTimeMillis() - time);
+      IStorageGroupMNode recoveredTree =
+          deserializeFrom(mtreeSnapshot, storageGroupMNode.getFullPath());
+      if (recoveredTree != null) {
+        this.storageGroupMNode.getParent().replaceChild(storageGroupMNode.getName(), recoveredTree);
+        this.storageGroupMNode = recoveredTree;
+        logger.debug(
+            "spend {} ms to deserialize mtree from snapshot", System.currentTimeMillis() - time);
+      }
     }
   }
 
+  public IStorageGroupMNode getStorageGroupMNode() {
+    return this.storageGroupMNode;
+  }
+
+  public void setStorageGroupMNode(IStorageGroupMNode storageGroupMNode) {
+    this.storageGroupMNode = storageGroupMNode;
+    levelOfSG = storageGroupMNode.getPartialPath().getNodeLength() - 1;
+  }
+
   public void clear() {
-    root = new InternalMNode(null, IoTDBConstant.PATH_ROOT);
+    storageGroupMNode = null;
   }
 
```

Comment:
Maybe we can add the SGM info in the log of createSnapshot()...

Reply:
The info of SGM and MTreeAboveSG is all recorded in the sg folders' names.